### ISSUE TITLE
[Mac] Save window positions of notebooks

### ISSUE THREAD

**Issue by dwt** [nteract/archived-desktop-app#5274]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated that I need to move the window of that specific notebook back to the side of the screen so I can see that other app that I need to enter data from. Would be much nicer if the app could just remember the window position of that notebook and restore it. I use this small data app I built in jupyter every day, so this is a daily chore of opening the thing, resizing it, putting it to the side of the screen, …

**Describe the solution you'd like**
When closing a notebook, it's window position could be stored. In Cocoa, this would entail calling a function like [setFrameAutoSaveName:](https://developer.apple.com/documentation/appkit/nswindow/1419509-setframeautosavename) with a unique name (something for example derived from the path of the notebook) which is not too hard.

**Describe alternatives you've considered**
Manual repositioning, boring and repetitive.

**Comment by willingc** [comment]
@dwt Thanks for the enhancement suggestion. I tend to use the Magnet app to position windows on my mac. Perhaps that may help you.

**Comment by dwt** [comment]
It probably will, but at the same time, this is a really nice Mac OS feature that especially document based apps like nteract can easily support and make the app much nicer for all users - even those not using crutches like Magnet.

**Comment by willingc** [comment]
@dwt I agree. If you have experience adding this functionality for macOS, we would welcome a PR. Thanks!

**Comment by captainsafia** [comment]
Good recommendation, indeed!

This GitHub issue (https://github.com/electron/electron/issues/526) has more information on how to do this and some sample implementations.

Tagging this as new-contributor-friendly since the code samples are pretty buildable.
